Tests/FluentAssertions.Specs/Specialized/TaskAssertionSpecs.cs

```diff
@@ -39,12 +42,12 @@ public async Task When_subject_is_null_it_should_throw()
                 timeSpan, "because we want to test the failure {0}", "message");
 
             // Assert
-            await testAction.Should().ThrowAsync<XunitException>()
+            await testAction.Should().ThrowWithinAsync<XunitException>(1.Seconds())
```

Comment:
Why are you changing many assertions to use `ThrowWithinAsync` and `CompleteWithinAsync`? 🤔

Reply:
See above.

> The only thing I've changed is using the assertions with timeout.
> This is reason why I've introduced them: The tested code may cause deadlocks. In this case the test may never complete.
> Asynchronous methods may always be tested with timeout.

Reply:
In what case can we deadlock?
Adding timeouts to combat deadlocks sounds like symptomatic treatment to me.

Reply:
Due to bugs in the tested code. This will create early failing tests.

Reply:
So we're talking about potential deadlocks?
I'd like to not add timeouts prematurely.

Reply:
For certain test suites it's undoubtedly useful, but let's leave them out of the this test suite.
Our tests are small enough that if we face deadlocks _something_ will fail and we can diagnose it at that time.
For now we think it makes the tests less clear on _exactly_ what scenario they're testing.

Reply:
I agree with @jnyrup here.

Reply:
done